### PR TITLE
Add an option to read in a list of points to scan from a .csv file

### PR DESCRIPTION
**Pull request by runtingt**
Adds option `--fromfile` to EnhancedCombine that allows the reading of points from a `.csv` file. Some checks for number of points and the requested algo (i.e. only 'fixed' is allowed) are performed.

### CODE REVIEW

**Comment by runtingt** [comment]
Documentation added in Combine docs: https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/937

**Comment by runtingt** [comment]
I think this should now be ready to merge, any suggestions for improvements?

File: CombineTools/python/combine/EnhancedCombine.py [hunk 1]
```diff
@@ -238,18 +269,48 @@ def run_method(self):
...
         # can only put the name option back now because we might have modified
         # it from what the user specified
         self.put_back_arg('name', '-n')
         proto = 'combine ' + (' '.join(self.passthru))
         if self.args.there:
             proto = 'pushd %(DIR)s; combine ' + (' '.join(self.passthru))+'; popd'
 
-        for it in itertools.product(*list(subbed_vars.values())):
+        job_str = ''
+        for idx, it in enumerate(itertools.product(*list(subbed_vars.values()))):
```
Comment (ajgilbert) [inline CombineTools/python/combine/EnhancedCombine.py:304]: Please avoid changing this part of the code, I don't think it's necessary - combineTool.py already has a mechanism to group individual combine calls into single jobs (`--merge X`).

**Review by ajgilbert — CHANGES_REQUESTED** [review]
Hi @runtingt, thanks for the PR. This is useful functionality, but I've added some comments that should be addressed first before we merge. Note also that combineTool.py already has a (not well documented) `--generate` option, which can be used to construct multiple jobs out of a series of options, e.g.:
```sh
> combineTool.py -M MultiDimFit --algo fixed --generate n:fixedPointPOIs::0,,r1=1,r2=2:1,,r1=2,r2=3 --dry-run
[DRY-RUN]: combine --algo fixed -M MultiDimFit --fixedPointPOIs r1=1,r2=2 -n .Test.0
[DRY-RUN]: combine --algo fixed -M MultiDimFit --fixedPointPOIs r1=2,r2=3 -n .Test.1
```
One alternative could be to simply append an extra argument to the internal list of generate options, e.g. 

```py
self.args.generate.append(
            "n:fixedPointPOIs::" + ":".join(["{},,".format(i) + ",".join(["{}={}".format(k,v) for k,v in X.items()]) for i, X in enumerate(pd.read_csv("test.csv").to_dict(orient="records"))]))
```
but also fine to stick with the current implementation.

**Comment by runtingt** [comment]
Hi @ajgilbert, thanks for the review and apologies for the delay. Since the scripts have now moved to Combine, I've actioned your comments there: https://github.com/cms-analysis/HiggsAnalysis-CombinedLimit/pull/937

**Pull request closed without merging**